In [1]:
# USOPC - Triathlon

import pandas as pd # library for data analysis
import requests # library to handle requests
from bs4 import BeautifulSoup # library to parse HTML documents
import numpy as np
from wikitable import wikitable # library to handle wikitables

In [2]:
# Get wikipedia tables

url = "https://en.wikipedia.org/wiki/Triathlon_at_the_2024_Summer_Olympics_%E2%80%93_Qualification#Men's"
summarydf = wikitable(url, to_csv=True, overwrite=True, table_number=1)
print(summarydf)

             NOC  Men  Women  Mixed  Total
0         France    2      2    NaN      4
1  Great Britain    2      2    NaN      4
2  Total: 2 NOCs    4      4    8.0      8


In [3]:
# Replace NaN with 1 for Yes/Qualified

summarydf = summarydf.replace(np.nan, 1)
print(summarydf)

             NOC  Men  Women  Mixed  Total
0         France    2      2    1.0      4
1  Great Britain    2      2    1.0      4
2  Total: 2 NOCs    4      4    8.0      8


In [4]:
# Get wikipedia tables

url = "https://en.wikipedia.org/wiki/Triathlon_at_the_2024_Summer_Olympics_%E2%80%93_Qualification#Men's"
eventsdf = wikitable(url, to_csv=True, overwrite=True, table_number=2)
print(eventsdf)

                                               Event                     Date  \
0     2022 World Triathlon Mixed Relay Championships            June 26, 2022   
1     2023 World Triathlon Mixed Relay Championships      July 15–16, 2023[4]   
2  2024 World Triathlon Mixed Relay Olympic Quali...  April 15 – May 27, 2024   
3         Mixed Relay Olympic Qualification Rankings           March 25, 2024   
4          Individual Olympic Qualification Rankings             May 27, 2024   

      Venue  
0  Montreal  
1   Hamburg  
2       TBA  
3         —  
4         —  


In [5]:
# Get wikipedia tables

url = "https://en.wikipedia.org/wiki/Triathlon_at_the_2024_Summer_Olympics_%E2%80%93_Qualification#Men's"
mensdf = wikitable(url, to_csv=True, overwrite=True, table_number=3)
print(mensdf)

                                                Event  Places  Qualified NOC  \
0                                         Host nation       2         France   
1                                         Host nation       2         France   
2      2022 World Triathlon Mixed Relay Championships       2  Great Britain   
3      2022 World Triathlon Mixed Relay Championships       2  Great Britain   
4      2023 World Triathlon Mixed Relay Championships       2            NaN   
5      2023 World Triathlon Mixed Relay Championships       2            NaN   
6                  Mixed Relay Qualification Rankings      12            NaN   
7                  Mixed Relay Qualification Rankings      12            NaN   
8                  Mixed Relay Qualification Rankings      12            NaN   
9                  Mixed Relay Qualification Rankings      12            NaN   
10                 Mixed Relay Qualification Rankings      12            NaN   
11                 Mixed Relay Qualifica

In [6]:
# Get wikipedia tables

url = "https://en.wikipedia.org/wiki/Triathlon_at_the_2024_Summer_Olympics_%E2%80%93_Qualification#Men's"
womensdf = wikitable(url, to_csv=True, overwrite=True, table_number=4)
print(womensdf)

                                                Event  Places  Qualified NOC  \
0                                         Host nation       2         France   
1                                         Host nation       2         France   
2      2022 World Triathlon Mixed Relay Championships       2  Great Britain   
3      2022 World Triathlon Mixed Relay Championships       2  Great Britain   
4      2023 World Triathlon Mixed Relay Championships       2            NaN   
5      2023 World Triathlon Mixed Relay Championships       2            NaN   
6                  Mixed Relay Qualification Rankings      12            NaN   
7                  Mixed Relay Qualification Rankings      12            NaN   
8                  Mixed Relay Qualification Rankings      12            NaN   
9                  Mixed Relay Qualification Rankings      12            NaN   
10                 Mixed Relay Qualification Rankings      12            NaN   
11                 Mixed Relay Qualifica

In [7]:
# Export tables as csv

summarydf.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\summarydf.csv')
eventsdf.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\eventsdf.csv')
mensdf.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\mensdf.csv')
womensdf.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\womensdf.csv')

In [8]:
# Import files from triathlon.com for verification Tableau

elite_men = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\OlympicEliteMen.csv')
elite_women = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\OlympicEliteWomen.csv')
mixedrelay = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\OlympicMixedRelay.csv')

elite_men.head()

Rank  Prior Rank Rank Change First Name Last Name   YOB Country  \
0     1           1           0        L�o   Bergere  1996     FRA   
1     2           2           0       Alex       Yee  1998     GBR   
2     3           3           0      Jelle     Geens  1993     BEL   
3     4           4           0    Vincent      Luis  1989     FRA   
4     5           6           1     Hayden     Wilde  1997     NZL   

   Total Points  Events First Period  Events Second Period  ...  \
0     4613.2000                    6                     0  ...   
1     4346.4100                    5                     0  ...   
2     3936.8545                    6                     0  ...   
3     3684.8900                    6                     0  ...   
4     3553.3200                    6                     0  ...   

   Period 1 5th Score  Period 1 6th Score  Period 1 7th Score  \
0              650.00            593.5900                 NaN   
1              732.09                 NaN                 NaN   
2              434.56            399.7245                 NaN   
3              371.82            343.9400                 NaN   
4              375.00             22.4600                 NaN   

   Period 2 Best Score  Period 2 2nd Score  Period 2 3rd Score  \
0                  NaN                 NaN                 NaN   
1                  NaN                 NaN                 NaN   
2                  NaN                 NaN                 NaN   
3                  NaN                 NaN                 NaN   
4                  NaN                 NaN                 NaN   

   Period 2 4th Score  Period 2 5th Score  Period 2 6th Score  \
0                 NaN                 NaN                 NaN   
1                 NaN                 NaN                 NaN   
2                 NaN                 NaN                 NaN   
3                 NaN                 NaN                 NaN   
4                 NaN                 NaN                 NaN   

   Period 2 7th Score  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 26 columns]

In [9]:
#only keep columns 'col1' and 'col2'
elite_men = elite_men[['Rank', 'First Name', 'Last Name', 'Country', 'Total Points']]

In [10]:
elite_women.head()

Rank  Prior Rank Rank Change First Name     Last Name   YOB Country  \
0     1           1           0     Taylor        Spivey  1991     USA   
1     2           2           0       Beth        Potter  1991     GBR   
2     3           3           0    Georgia  Taylor-Brown  1994     GBR   
3     4           4           0  Cassandre     Beaugrand  1997     FRA   
4     5           5           0      Flora         Duffy  1987     BER   

   Total Points  Events First Period  Events Second Period  ...  \
0     4135.4400                    6                     0  ...   
1     4092.8800                    6                     0  ...   
2     3976.7900                    5                     0  ...   
3     3666.1245                    6                     0  ...   
4     3469.8000                    4                     0  ...   

   Period 1 5th Score  Period 1 6th Score  Period 1 7th Score  \
0              593.59            593.5900                 NaN   
1              549.07            495.7600                 NaN   
2              251.79                 NaN                 NaN   
3              507.89            399.7245                 NaN   
4                 NaN                 NaN                 NaN   

   Period 2 Best Score  Period 2 2nd Score  Period 2 3rd Score  \
0                  NaN                 NaN                 NaN   
1                  NaN                 NaN                 NaN   
2                  NaN                 NaN                 NaN   
3                  NaN                 NaN                 NaN   
4                  NaN                 NaN                 NaN   

   Period 2 4th Score  Period 2 5th Score  Period 2 6th Score  \
0                 NaN                 NaN                 NaN   
1                 NaN                 NaN                 NaN   
2                 NaN                 NaN                 NaN   
3                 NaN                 NaN                 NaN   
4                 NaN                 NaN                 NaN   

   Period 2 7th Score  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  

[5 rows x 26 columns]

In [11]:
#only keep columns 'col1' and 'col2'
elite_women = elite_women[['Rank', 'First Name', 'Last Name', 'Country', 'Total Points']]

In [12]:
mixedrelay.head()

Rank  Prior Rank  Rank Change             Team Name Country  Total Points  \
0     1           1            0  Team I Great Britain     GBR       2465.00   
1     2           2            0         Team I France     FRA       2334.50   
2     3           3            0        Team I Germany     GER       2085.75   
3     4           4            0    Team I New Zealand     NZL       1917.13   
4     5           5            0    Team I Switzerland     SUI       1866.51   

   Points Current Period  Points Previous Period  Events Current Period  \
0                      0                 2465.00                      0   
1                      0                 2334.50                      0   
2                      0                 2085.75                      0   
3                      0                 1917.13                      0   
4                      0                 1866.51                      0   

   Events Previous Period  Current Best Score  Current 2nd Score  \
0                       3                 NaN                NaN   
1                       3                 NaN                NaN   
2                       3                 NaN                NaN   
3                       3                 NaN                NaN   
4                       3                 NaN                NaN   

   Current 3rd Score  Previous Best Score  Previous 2nd Score  \
0                NaN               925.00              800.00   
1                NaN              1000.00              684.50   
2                NaN               800.00              684.50   
3                NaN               791.45              585.68   
4                NaN               677.19              633.16   

   Previous 3rd Score  
0              740.00  
1              650.00  
2              601.25  
3              540.00  
4              556.16

In [13]:
#only keep columns 'col1' and 'col2'
mixedrelay = mixedrelay[['Rank', 'Team Name', 'Country', 'Total Points']]

In [14]:
# Import files from triathlon.com for verification Tableau

world_men = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\WorldRankingsEliteMen.csv')
world_women = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\WorldRankingsEliteWomen.csv')
worldmixedrelay = pd.read_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\MixedRelayMixedRelay.csv')

#only keep columns 'col1' and 'col2'
world_men = world_men[['Rank', 'First Name', 'Last Name', 'Country', 'Total Points']]
world_men.head()

Rank First Name Last Name Country  Total Points
0     1       Alex       Yee     GBR       6386.37
1     2        L�o   Bergere     FRA       6243.86
2     3     Hayden     Wilde     NZL       5442.82
3     4    Vincent      Luis     FRA       4759.69
4     5      Jelle     Geens     BEL       4632.83

In [15]:
#only keep columns 'col1' and 'col2'
world_women = world_women[['Rank', 'First Name', 'Last Name', 'Country', 'Total Points']]
world_women.head()

Rank First Name     Last Name Country  Total Points
0     1      Flora         Duffy     BER       5545.76
1     2    Georgia  Taylor-Brown     GBR       5405.95
2     3     Taylor        Spivey     USA       5337.53
3     4       Beth        Potter     GBR       4980.99
4     5     Taylor         Knibb     USA       4700.70

In [16]:
#only keep columns 'col1' and 'col2'
worldmixedrelay = worldmixedrelay[['Rank', 'Team Name', 'Country', 'Total Points']]
worldmixedrelay.head()

Rank             Team Name Country  Total Points
0     1  Team I Great Britain     GBR       2930.84
1     2        Team I Germany     GER       2763.89
2     3         Team I France     FRA       2586.15
3     4    Team I New Zealand     NZL       2428.85
4     5    Team I Switzerland     SUI       2269.04

In [17]:
#filter table for US athletes (top 30(qualification) and top 140(world))
# How many US atlhetes are currently on track to qualify?

# selecting rows based on condition 
us_men_rankings = world_men[world_men['Country'] == 'USA'] 
    
print('\nResult dataframe :\n', us_men_rankings)


Result dataframe :
       Rank First Name       Last Name Country  Total Points
11      12    Matthew         Mcelroy     USA       2984.55
27      28       Seth           Rider     USA       2230.05
47      48     Morgan         Pearson     USA       1802.39
51      52      Kevin        Mcdowell     USA       1738.09
56      57      Chase         McQueen     USA       1673.21
71      72       Darr           Smith     USA       1492.70
124    125       John            Reed     USA        958.93
143    144       Marc         Dubrick     USA        861.44
145    146      Brent        Demarest     USA        858.93
168    169     Austin         Hindman     USA        768.72
175    176      Ka'eo           Kruse     USA        742.03
180    181   Nicholas          Holmes     USA        729.18
193    194     Andrew   Shellenberger     USA        671.52
197    198      Davis            Bove     USA        639.36
200    201    Michael        Arishita     USA        631.03
270    271     Thom

### zero qualified
### seven on track

##        Rank First Name       Last Name Country  Total Points
### 11      12    Matthew         Mcelroy     USA       2984.55
### 27      28       Seth           Rider     USA       2230.05
### 47      48     Morgan         Pearson     USA       1802.39
### 51      52      Kevin        Mcdowell     USA       1738.09
### 56      57      Chase         McQueen     USA       1673.21
### 71      72       Darr           Smith     USA       1492.70
### 124    125       John            Reed     USA        958.93

In [18]:
#filter table for US athletes (top 30(qualification) and top 140(world))
# How many US atlhetes are currently on track to qualify?

# selecting rows based on condition 
us_men_elite_rankings = elite_men[elite_men['Country'] == 'USA'] 
    
print('\nResult dataframe :\n', us_men_elite_rankings)


Result dataframe :
      Rank First Name       Last Name Country  Total Points
8       9    Matthew         McElroy     USA       2648.18
31     32       Seth           Rider     USA       1467.44
41     42     Morgan         Pearson     USA       1164.47
68     69      Kevin        Mcdowell     USA        741.14
69     70       Darr           Smith     USA        718.74
71     72      Chase         McQueen     USA        697.53
134   135      Brent        Demarest     USA        228.59
168   169    Michael        Arishita     USA        131.87
188   189     Austin         Hindman     USA         69.28
205   206     Thomas  Sonnery-Cottet     USA         48.87
206   207      Davis            Bove     USA         45.49
213   214      Griff          Morgan     USA         30.56
223   224     Andrew   Shellenberger     USA         20.95
235   236       Ryan            Lund     USA         12.14
241   241       Marc         Dubrick     USA          8.22


### zero qualified
### one on track

##      Rank First Name       Last Name Country  Total Points
### 8       9    Matthew         McElroy     USA       2648.18

In [19]:
#filter table for US athletes (top 30(qualification) and top 140(world))
# How many US atlhetes are currently on track to qualify?

# selecting rows based on condition 
us_women_rankings = world_women[world_women['Country'] == 'USA'] 
    
print('\nResult dataframe :\n', us_women_rankings)


Result dataframe :
      Rank First Name   Last Name Country  Total Points
2       3     Taylor      Spivey     USA       5337.53
4       5     Taylor       Knibb     USA       4700.70
14     15    Kirsten      Kasper     USA       3084.80
15     16     Summer   Rappaport     USA       3054.02
28     29      Erika   Ackerlund     USA       2317.10
44     45       Gina      Sereno     USA       1772.11
94     95      Katie     Zaferes     USA       1196.56
148   149  Gabrielle      Lumkes     USA        808.16
176   177      Lydia     Russell     USA        669.61
209   210    Madisen       Lavin     USA        532.20
220   221       Gwen   Jorgensen     USA        498.80
222   223      Grace     Walther     USA        489.29
224   225     Tamara      Gorman     USA        479.95
255   256      Annie       Kelly     USA        362.63
260   261      Faith       Dasso     USA        351.04
267   268      Annie      Fuller     USA        337.67
277   278   Michelle    Stratton     USA    

### zero qualified
### seven on track

##       Rank First Name   Last Name Country  Total Points
### 2       3     Taylor      Spivey     USA       5337.53
### 4       5     Taylor       Knibb     USA       4700.70
### 14     15    Kirsten      Kasper     USA       3084.80
### 15     16     Summer   Rappaport     USA       3054.02
### 28     29      Erika   Ackerlund     USA       2317.10
### 44     45       Gina      Sereno     USA       1772.11
### 94     95      Katie     Zaferes     USA       1196.56

In [20]:
#filter table for US athletes (top 30(qualification) and top 140(world))
# How many US atlhetes are currently on track to qualify?

# selecting rows based on condition 
us_women_elite_rankings = elite_women[elite_women['Country'] == 'USA'] 
    
print('\nResult dataframe :\n', us_women_elite_rankings)


Result dataframe :
      Rank First Name  Last Name Country  Total Points
0       1     Taylor     Spivey     USA       4135.44
9      10     Taylor      Knibb     USA       2769.95
14     15     Summer  Rappaport     USA       2222.20
23     24    Kirsten     Kasper     USA       1791.88
30     31      Erika  Ackerlund     USA       1562.41
63     64       Gina     Sereno     USA        831.86
107   108     Tamara     Gorman     USA        303.91
142   143       Gwen  Jorgensen     USA        136.10
177   178      Katie    Zaferes     USA         45.31
178   179      Annie      Kelly     USA         42.47
183   184    Madisen      Lavin     USA         33.45


### zero qualified
### four on track

##      Rank First Name  Last Name Country  Total Points
### 0       1     Taylor     Spivey     USA       4135.44
### 9      10     Taylor      Knibb     USA       2769.95
### 14     15     Summer  Rappaport     USA       2222.20
### 23     24    Kirsten     Kasper     USA       1791.88

In [21]:
# Export tables as csv

us_men_rankings.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\us_men_rankings.csv')
us_women_rankings.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\us_women_rankings.csv')
us_men_elite_rankings.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\us_men_elite_rankings.csv')
us_women_elite_rankings.to_csv(r'C:\Users\andyr\OneDrive\Desktop\USOPC\us_women_elite_rankings.csv')